backend/app/api/handlers/v1/controller.go

```diff
@@ -144,11 +145,40 @@ func (ctrl *V1Controller) HandleCurrency() errchain.HandlerFunc {
 }
 
 func (ctrl *V1Controller) HandleCacheWS() errchain.HandlerFunc {
+	type eventMsg struct {
+		Event string `json:"event"`
+	}
+
 	m := melody.New()
 
 	m.HandleConnect(func(s *melody.Session) {
 		auth := services.NewContext(s.Request.Context())
 		s.Set("gid", auth.GID)
+
+		// Asynchronous ticker that keeps the websocket connection alive with periodic pings.
+		go func() {
+			const interval = 10 * time.Second
+
+			ping := time.NewTicker(interval)
+			defer ping.Stop()
+
+			for {
+				select {
+				case <-s.Request.Context().Done():
+					return
+
+				case <-ping.C:
+					msg := &eventMsg{Event: "ping"}
+
+					pingBytes, err := json.Marshal(msg)
+					if err != nil {
+						log.Log().Msgf("error marshaling ping: %v", err)
+					} else {
+						_ = m.BroadcastMultiple(pingBytes, []*melody.Session{s})
+					}
+				}
+			}
+		}()
```

Comment:
Instead of creating a new go routine every connection, could we ping on a constant interval to any open connection instead? That way we'd only have the one go routine to worry about.